src/mem/pagemap.h

```diff
@@ -168,6 +198,31 @@ namespace snmalloc
     }
 
   public:
+    /**
+     * The pagemap configuration describing this instantiation of the template.
+     */
+    static constexpr PagemapConfig config = {
+      1, false, GRANULARITY_BITS, sizeof(T)};
+
+    /**
+     * Cast a `void*` to a pointer to this template instantiation, given a
+     * config describing the configuration.  Return null if the configuration
+     * passed does not correspond to this template instantiation.
+     *
+     * This intended to allow code that depends on the pagemap having a
+     * specific representation to fail gracefully.
+     */
+    static Pagemap* cast_to_pagemap(void* pm, const PagemapConfig* c)
+    {
```

Comment:
Check for null config and succeed?

Reply:
I don't see when that would be useful.  If you have a null config, it's only because you haven't asked for one. You can just do a `static_cast` if you don't want any type checking.